packages/adapter-components/src/definitions/system/fetch/element.ts

```diff
@@ -56,6 +56,8 @@ export type ElemIDDefinition<TCustomNameMappingOptions extends string = never> =
 export type PathDefinition<TCustomNameMappingOptions extends string = never> = {
   // when id parts info is missing, inherited from elemID (but the values are path-nacl-cased)
   pathParts?: IDPartsDefinition<TCustomNameMappingOptions>[] & { extendsParent?: boolean }
+  // allow adding hard coded prefix for path that will be added before the type name
+  prefix?: string
```

Comment:
What do you think about allowing multiple levels of nesting?
```suggestion
  prefix?: string[]
```

Reply:
I think it's better not to use `prefix` here as it's usually used inside a specific string. maybe `baseDir` or `root`?

Reply:
also why not add this next to `extendsParent`?

Reply:
`extendsParent` is on the elemID and not on path ?

Reply:
my bad, I see it's also defined on the path, but in each element of the array
can you please remind me why `pathParts` is an array of `parts` ?

Reply:
oops I see I misread the definition of `extendsParent` - so it's the other way around, we should move it to be next to this and *not* on `pathParts`, I see this was adde by @edenhassid  in https://github.com/salto-io/salto/pull/6115 with your review so I'm not sure I fully follow why it was defined next to the array this way, maybe one of you will remember? we don't have to fix it as part of this PR but I don't think it's usable the way it's defined, there's no way to do this in json even if typescript allows it.

the path contains multiple folders (`a/b/c`) and each path part defines a folder, which itself can be built based on parts similarly to the elem id (`c = x_y_z`)

Reply:
thanks, IIRC I think the use case for  `pathBaseDir` to be under parts is to allow flexibility in each part to extend the parent path and it was done intentionally
I kind of agree this is a bit strange and we might want to change it

anyway I think `baseDir` should be directly under `PathDefinition`

Reply:
@edenhassid is there a reason to keep it "next to the array" the way it's defined now? or was that a mistake and we can move it to be next to the new param? :)

Reply:
it wasn't a mistake, I thought at the time that there is no reason to limit this only to the file name, and to add the possibility to support this for folders as well.